### PR TITLE
use correct mappings when they exist

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug

> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
a rebase failure between infra container and userns caused the wrong mappings to be used
This could cause a segfault, and causes wrong behavior. Fix it

Signed-off-by: Peter Hunt <pehunt@redhat.com>
#### Does this PR introduce a user-facing change?

```release-note
fix bug where we attempted to chown with the mappings configured on server level, when they could have been from annotations
```

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
LGTM

File: server/sandbox_run_linux.go [hunk 1]
```diff
@@ -763,7 +763,7 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 			}
 			g.AddMount(proc)
 		}
-		rootPair := s.defaultIDMappings.RootPair()
+		rootPair := sandboxIDMappings.RootPair()
```
Comment (TomSweeneyRedHat) [inline server/sandbox_run_linux.go:766]: based on the title, I expected to see something like:
```suggestion
		rootPair := sandboxIDMappings.RootPair(); rootPair != nil {
		    rootPair = s.defaultIDMappings.RootPair()
		}
```

Comment (haircommander) [reply server/sandbox_run_linux.go:766]: the segfault came from s.defaultIDMappings being nil, rootPair is an object, not a pointer :)

**Pull request merged**